src/mitigation.c

```diff
@@ -54,18 +56,37 @@ void mitigation_add_resource(struct mitigation *m, const char *name, const char
 
 	r->resource = res;
 
-	list_append(&m->resources, &r->list_node);
+	// resources that have the attribute fallback will be added to a seperate list
+	if(fallback)
+		list_append(&m->fallback_resources, &r->list_node);
+	else
+		list_append(&m->resources, &r->list_node);
 }
 
 void mitigation_activate(struct mitigation *m)
 {
 	struct mitigation_resource *r;
 	struct list_node *node;
+	int use_fallback_resources;
+	int ret;
 
+	use_fallback_resources = 0;
 	for_list_node(&m->resources, node) {
 		r = list_entry(node, struct mitigation_resource, list_node);
 		resource_enable(r->resource);
-		resource_write_value(r->resource, r->target_value, strlen(r->target_value));
+		ret = resource_write_value(r->resource, r->target_value, strlen(r->target_value));
+		// for an error in resource_write_value, declare the need to use fallback_resources
+		if(ret)
+			use_fallback_resources = 1;
```

Comment:
The latter is actually what i was targeting, so a fallback to the whole mitigation. The reason i don't disable failed resources is because that wasn't there before. The fallback basically comes as a second layer protection after the normal behavior (pre-fallback) finishes.